src/java.base/share/classes/sun/security/ssl/SSLEngineImpl.java

```diff
@@ -1270,7 +1270,12 @@ public Void run() throws Exception {
                     Map.Entry<Byte, ByteBuffer> me =
                             context.delegatedActions.poll();
                     if (me != null) {
-                        context.dispatch(me.getKey(), me.getValue());
+                        try {
+                            context.dispatch(me.getKey(), me.getValue());
+                        } catch (Exception e) {
+                            throw context.conContext.fatal(Alert.INTERNAL_ERROR,
+                                    "Unhandled exception", e);
+                        }
```

Comment:
Is there any chance this will double alert?

Reply:
Not really. Only the first call to `fatal` sends an alert, subsequent calls just rethrow the given exception, wrapping it in a SSLException if necessary.

Reply:
Thanks for confirming.